### PR TITLE
Added support for my new FeatherS2 Neo

### PR DESCRIPTION
**Pull request by UnexpectedMaker**
Also remembered to dded FeatherS2 Neo to build script this time :)

VID/PID is from Espressif:
https://github.com/espressif/usb-pids/blob/main/allocated-pids.txt

Cheers :)

### CODE REVIEW

File: .github/workflows/build_esp32.yml [hunk 1]
```diff
@@ -40,6 +40,7 @@ jobs:
         - 'targett_mcb_wrover'
         - 'unexpectedmaker_feathers2'
         - 'unexpectedmaker_tinys2'
+        - 'unexpectedmaker_feathers2_neo'
```
Comment (hathach) [inline .github/workflows/build_esp32.yml:43]: ABC order, should move 1 line up

File: ports/esp32s2/boards/unexpectedmaker_feathers2_neo/board.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+// USB UF2
+//--------------------------------------------------------------------+
+
+#define USB_VID           0x303A
+#define USB_PID           0x80B5
+#define USB_MANUFACTURER  "Unexpected Maker"
+#define USB_PRODUCT       "FeatherS2 Neo"
+
+#define UF2_PRODUCT_NAME  USB_MANUFACTURER " " USB_PRODUCT
+#define UF2_BOARD_ID      "UnexpectedMaker-FeatherS2-Neo"
```
Comment (hathach) [inline ports/esp32s2/boards/unexpectedmaker_feathers2_neo/board.h:67]: board id must respect the uf2 spec https://github.com/microsoft/uf2#files-exposed-by-bootloaders

**Review by hathach — CHANGES_REQUESTED** [review]
Board id must follow the uf2 specs, please correct

```
The Board-ID field is machine-readable and consists of a number of dash-separated tokens. The first token is the CPU type, second is the board type, and third is the board revision. More tokens can be also added.
```

File: ports/esp32s2/boards/unexpectedmaker_feathers2_neo/board.h (hunk 2 above)
Comment (UnexpectedMaker) [reply ports/esp32s2/boards/unexpectedmaker_feathers2_neo/board.h:67]: When did this change of requirement happen?
None of my previous boards required this and most boards in the port don't follow that spec.

Is this then going to be enforced on all previous boards?

File: .github/workflows/build_esp32.yml (hunk 1 above)
Comment (UnexpectedMaker) [reply .github/workflows/build_esp32.yml:43]: Sorry, I didn't realise it was alphabetical beyond the starting word.

File: ports/esp32s2/boards/unexpectedmaker_feathers2_neo/board.h (hunk 2 above)
Comment (UnexpectedMaker) [reply ports/esp32s2/boards/unexpectedmaker_feathers2_neo/board.h:67]: Sorry, I stand corrected - most are in this format now, just mine and a few others are not.
Maybe I missed this change of requirements. I'll fix my boards now.

**Review by hathach — APPROVED** [review]
great! Thank you for your pr, yeah I try to enforce the board id specs since it is common mistake from user. Sometimes I forgot about it as well.

**Pull request merged**